src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/barcode/BarcodeEventRepository.kt

```diff
@@ -0,0 +1,57 @@
+package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.barcode
+
+import org.hibernate.Hibernate
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.stereotype.Repository
+import java.time.LocalDateTime
+import javax.persistence.Entity
+import javax.persistence.EnumType
+import javax.persistence.Enumerated
+import javax.persistence.GeneratedValue
+import javax.persistence.GenerationType
+import javax.persistence.Id
+import javax.persistence.JoinColumn
+import javax.persistence.ManyToOne
+import javax.persistence.Table
+import javax.validation.constraints.NotNull
+
+@Repository
+interface BarcodeEventRepository : JpaRepository<BarcodeEvent, Long> {
+  fun findByBarcode(barcode: Barcode): List<BarcodeEvent>
+}
+
+@Entity
+@Table(name = "barcode_events")
+data class BarcodeEvent(
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  @NotNull
+  val id: Long = -1,
+  @NotNull
+  @ManyToOne
+  @JoinColumn(name = "barcode")
+  val barcode: Barcode,
+  @NotNull
+  val userId: String,
+  @NotNull
+  @Enumerated(EnumType.STRING)
+  val status: BarcodeStatus = BarcodeStatus.CREATED,
+  @NotNull
+  val dateTime: LocalDateTime = LocalDateTime.now(),
+) {
+  override fun equals(other: Any?): Boolean {
```

Comment:
In general we want equals and hashcode to match the business key which is barcode / date time because entities are mutable by their nature... however in this case the entity is actually immutable by design so maybe you're right. But on the other hand implementing equals makes it explicitly clear what the business key is. Hmmm need to think on this one some more.